release/ray_release/tests/test_test.py

```diff
@@ -264,5 +278,35 @@ def test_is_high_impact() -> None:
     assert not _stub_test({"name": "test"}).is_high_impact()
 
 
+@patch("ray_release.test.Test._gen_test_result")
+def test_gen_test_results(mock_gen_test_result) -> None:
```

Comment:
I think pytest supports async test functions, probably needs to import a plugin package though.

Reply:
it requires an extra dependency https://pypi.org/project/pytest-asyncio/ so it's simple enough to handle the async function myself so i go with this solution for now; let me know if you prefer the other plugin